aliyun-ecs/src/test/java/org/jclouds/aliyun/ecs/compute/ECSComputeServiceLiveTest.java

```diff
@@ -0,0 +1,113 @@
+import org.jclouds.rest.AuthorizationException;
+import org.jclouds.scriptbuilder.domain.Statement;
+import org.jclouds.scriptbuilder.domain.Statements;
+import org.jclouds.scriptbuilder.statements.java.InstallJDK;
+import org.jclouds.scriptbuilder.statements.login.AdminAccess;
+import org.jclouds.sshj.config.SshjSshClientModule;
+import org.testng.annotations.Test;
+
+import java.util.Properties;
+
+import static org.jclouds.aliyun.ecs.compute.options.ECSServiceTemplateOptions.Builder.vpcId;
+
+/**
+ * Live tests for the {@link org.jclouds.compute.ComputeService} integration.
+ */
+@Test(groups = "live", singleThreaded = true, testName = "ECSComputeServiceLiveTest")
+public class ECSComputeServiceLiveTest extends BaseComputeServiceLiveTest {
+
+   private String vpcId;
+   private String vSwitchId;
+
+   public ECSComputeServiceLiveTest() {
+      provider = "aliyun-ecs";
+   }
+
+   @Override
+   protected Properties setupProperties() {
+      Properties properties = super.setupProperties();
+      vpcId = setIfTestSystemPropertyPresent(properties,  provider + ".vpcId");
+      vSwitchId = setIfTestSystemPropertyPresent(properties,  provider + ".vSwitchId");
+      return properties;
+   }
+
+   @Override
+   protected TemplateBuilder templateBuilder() {
+      return super.templateBuilder()
+                      .options(vpcId(vpcId)
+                      .vSwitchId(vSwitchId));
+   }
+
+   @Override
+   protected Template addRunScriptToTemplate(Template template) {
+      template.getOptions().runScript(Statements.newStatementList(
+            new Statement[] { AdminAccess.standard(), Statements.exec("sleep 50"), InstallJDK.fromOpenJDK() }));
+      return template;
+   }
+
+   @Override
+   protected Module getSshModule() {
+      return new SshjSshClientModule();
+   }
+
+   @Override
+   @Test(expectedExceptions = AuthorizationException.class)
+   public void testCorrectAuthException() throws Exception {
+      ComputeServiceContext context = null;
+      try {
+         Properties overrides = setupProperties();
+         overrides.setProperty(provider + ".identity", "MOM:MA");
+         overrides.setProperty(provider + ".credential", "MIA");
+         context = newBuilder().modules(ImmutableSet.of(getLoggingModule(), credentialStoreModule)).overrides(overrides)
+               .build(ComputeServiceContext.class);
+         // replace listNodes with listImages as it doesn't require `projectId`
+         context.getComputeService().listImages();
+      } catch (AuthorizationException e) {
+         throw e;
+      } catch (RuntimeException e) {
+         e.printStackTrace();
+         throw e;
+      } finally {
+         if (context != null)
+            context.close();
+      }
+   }
+
+   @Override
+   public void testOptionToNotBlock() throws Exception {
+      // Aliyun ECS ComputeService implementation has to block until the node
+      // is provisioned, to be able to return it.
```

Comment:
If we don't block after power on, we can probably remove this override. It would be a nice to have, as jclouds also provides non-blocking options when deploying nodes.